cognite/client/data_classes/transformations/__init__.py

```diff
@@ -148,11 +150,14 @@ def __init__(
         self.is_public = is_public
         self.ignore_null_fields = ignore_null_fields
         self.source_oidc_credentials = source_oidc_credentials
-        self.has_source_oidc_credentials = has_source_oidc_credentials or source_oidc_credentials is not None
+        if has_source_oidc_credentials or has_destination_oidc_credentials:
+            warnings.warn(
+                "The arguments 'has_source_oidc_credentials' and 'has_destination_oidc_credentials' are "
+                "deprecated and will be removed in a future version."
+                "These are now properties returning whether the transformation has source or destination oidc credentials set.",
+                DeprecationWarning,
```

Comment:
I vote for UserWarning, as 99/100 won't see the (correct) `DeprecationWarning`. Up to you